### PR TITLE
feat: register public key and declare chain support commands

### PR DESCRIPTION
**Pull request by haiyizxx**
## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: ampd/src/commands/declare_chain_support.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        contract: config.service_registry.cosmwasm_contract.as_ref().clone(),
+        msg,
+        funds: vec![],
+    }
+    .into_any()
+    .expect("failed to serialize proto message");
+
+    Ok(Some(format!(
+        "successfully broadcast declare chain support transaction, tx hash: {}",
+        broadcast_tx(config, tx, pub_key).await?.txhash
```
Comment (cgorenflo) [inline ampd/src/commands/declare_chain_support.rs:42]: I'd prefer to assign a variable for this for better readability

File: ampd/src/commands/register_public_key.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        .iter()
+        .find_map(|config| {
+            if let handlers::config::Config::MultisigSigner { cosmwasm_contract } = config {
+                Some(cosmwasm_contract.clone())
+            } else {
+                None
+            }
+        })
+        .ok_or(Error::LoadConfig)
+        .attach_printable("no multisig contract found in config")?;
```
Comment (cgorenflo) [inline ampd/src/commands/register_public_key.rs:30]: does it make sense to make finding a specific config a function on the handler `Config`? Also, at this point it seems a bit weird to call these handler configs

File: ampd/src/commands/register_public_key.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        contract: multisig_address.as_ref().clone(),
+        msg,
+        funds: vec![],
+    }
+    .into_any()
+    .expect("failed to serialize proto message");
+
+    Ok(Some(format!(
+        "successfully broadcast register public key transaction, tx hash: {}",
+        broadcast_tx(config, tx, pub_key).await?.txhash
```
Comment (cgorenflo) [inline ampd/src/commands/register_public_key.rs:49]: extract a variable for better readability

File: ampd/src/commands/bond_worker.rs [hunk 4]
```diff
@@ -40,7 +40,7 @@ pub async fn run(config: Config, state_path: &Path, args: Args) -> Result<Option
     .expect("failed to serialize proto message");
 
     Ok(Some(format!(
-        "successfully broadcasted bond worker transaction, tx hash: {}",
```
Comment (fish-sammy) [inline ampd/src/commands/bond_worker.rs:43]: What's wrong with "broadcasted"?

File: ampd/src/commands/register_public_key.rs (hunk 2 above)
Comment (fish-sammy) [reply ampd/src/commands/register_public_key.rs:30]: I also agree that it's weird to pull the handler config here.

**Review by fish-sammy — APPROVED** [review]

File: ampd/src/commands/bond_worker.rs (hunk 4 above)
Comment (haiyizxx) [reply ampd/src/commands/bond_worker.rs:43]: the past tense is same as present 😅

File: ampd/src/commands/register_public_key.rs (hunk 2 above)
Comment (haiyizxx) [reply ampd/src/commands/register_public_key.rs]: it's weird, can't find a good way atm. extracted to a helper function for now

**Pull request merged**